### PR TITLE
🔙 Updated BackHandler Docs with snack examples 🤾🏽‍♂️

### PR DESCRIPTION
**Pull request by DaniAkash**
Belongs to https://github.com/facebook/react-native-website/issues/1579

Have added snack examples for BackHandler and also added references to React Navigation & React Native hooks.

### CODE REVIEW

File: docs/backhandler.md [hunk 1]
```diff
@@ -3,62 +3,166 @@ id: backhandler
...
+      ]);
+      return true;
+    };
+
+    const backHandler = BackHandler.addEventListener(
+      "hardwareBackPress",
+      backAction
+    );
+
+    return () => backHandler.remove();
```
Comment (rachelnabors) [inline docs/backhandler.md]: Question: should this example use `Backhandler.removeEventListener()`?

Comment (DaniAkash) [reply docs/backhandler.md]: `removeEventListener` is a static method that requires passing the reference of the event handler to clear the event listener. 
However, `addEventListener` already returns a `NativeEventSubscription`
Screenshot 2020-03-04 at 4 31 12 AM
which can be cleared by calling the remove method which feels more straightforward 
Screenshot 2020-03-04 at 4 31 24 AM

If we need to explain the usage of `removeEventListener` it should be a separate example

Comment (rachelnabors) [reply docs/backhandler.md]: I feel like that needs covering as well. Ideally the example should include all of the API—or have text explaining where and how to use the bits that aren't covered.

Comment (DaniAkash) [reply docs/backhandler.md]: Seems like I need to come up with an explanation & an example on this 🤔

Comment (rachelnabors) [reply docs/backhandler.md]: Would you please, @DaniAkash? Would really complete the docs!

Comment (DaniAkash) [reply docs/backhandler.md]: Added the explanation!

**Pull request merged**